### DISCUSSION TITLE
I’ve been learning Python for a couple of years know and I was wondering how I could make my login system more secure when saving data

### DISCUSSION THREAD

**Discussion by ghost** [community/community#60964] [Programming Help]
### Body

[Click this link to go to the project](https://github.com/pytutorials/PythonTerminal/edit/main/Login.py)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ghost** [comment]
Also, I don’t know how to use a database with Python

**Comment by noahkaboa** [comment]
A common practice when storing passwords is to [hash](https://cs.stackexchange.com/questions/55471/what-exactly-and-precisely-is-hash) them. This way, even if an attacker is able to access the database, they won't be able to discern the users passwords. This can be implemented by storing the hashed passwords in your database, and then comparing to the hash stored in the database. Because of hashes anti-collision (no two inputs map to the same output) property, this allows for checking the password without revealing it.  
As for your  database question, I recommend using MongoDB, which is easy to work with in Python and is free. There are a ton of tutorials online for how to use it.
